src/scripting/context.rs

```diff
@@ -24,10 +26,15 @@ use tokio::time::Instant;
 use tracing::error;
 use try_lock::TryLock;
 
+static IS_SELECT_QUERY: Lazy<Regex> = Lazy::new(|| Regex::new(r"(?i)^\s*select\b").unwrap());
+static IS_SELECT_COUNT_QUERY: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"(?i)^\s*select\s+count\s*\(\s*[^)]*\s*\)").unwrap());
+
 #[derive(Clone, Debug, PartialEq)]
 pub struct PartitionGroup {
     pub n_rows_per_group: u64,
     pub n_partitions: u64,
+    pub partition_size: u64,
```

Comment:
if partition size means partition rows count, then better to name it `partition_rows_count`

Reply:
Will update